compiler/gradient_with_order.cc

```diff
@@ -132,18 +159,41 @@ void AddGradientNodesForTrainingWithOrders(Graph* graph, const std::vector<Order
 
     auto schedule_node = [&schedule_recompute](Node* node) { schedule_recompute(node, node); };
 
-    {
-        ScheduleAddedScope schedule_scope(graph, schedule_node);
-        SetInitialGradients(graph);
+    if (fwd_graph == bwd_graph) {
+        ScheduleAddedScope schedule_scope(fwd_graph, schedule_node);
+        SetInitialGradients(fwd_graph);
     }
 
     std::set<Node*> scheduled_forward;
+    Graph* current_graph = fwd_graph;
+
     for (const Order& order : orders) {
+        // Check if we should turn to the backward part
+        if (fwd_graph != bwd_graph && current_graph == fwd_graph) {
+            bool output_all_staged = true;
+            for (Value* output : fwd_graph->output_values()) {
+                if (!staged.count(output)) output_all_staged = false;
+            }
+            if (output_all_staged) {
+                current_graph = bwd_graph;
+                {
+                    ScheduleAddedScope schedule_scope(fwd_graph, schedule_node);
+                    {
```

Comment:
No need for the extra nest?

Reply:
Without this nest, I could not compile the code. 

```
[ 86%] Building CXX object compiler/CMakeFiles/chainer_compiler_compiler.dir/gradient_with_order.cc.o
/home/mkusumoto/chainer-compiler/compiler/gradient_with_order.cc: In function ‘void chainer_compiler::AddGradientNodesForTrainingWithOrders(chainer_compiler::Graph*, chainer_compiler::Graph*, const std::vector<chainer_compiler::Order>&)’:
/home/mkusumoto/chainer-compiler/compiler/gradient_with_order.cc:181:59: error: redeclaration of ‘chainer_compiler::{anonymous}::ScheduleAddedScope schedule_scope’
                         ScheduleAddedScope schedule_scope(bwd_graph, schedule_node);
                                                           ^
/home/mkusumoto/chainer-compiler/compiler/gradient_with_order.cc:180:40: note: ‘chainer_compiler::{anonymous}::ScheduleAddedScope schedule_scope’ previously declared here
                     ScheduleAddedScope schedule_scope(fwd_graph, schedule_node);

```

Reply:
Renaming variables should work?
```
ScheduleAddedScope fwd_schedule_scope(fwd_graph, schedule_node);
ScheduleAddedScope bwd_schedule_scope(bwd_graph, schedule_node);
```

Reply:
Yes, you are true. (sorry for silly mistake...) Fixed.